runtime/cudaq/qis/state.cpp

```diff
@@ -111,6 +111,33 @@ extern "C" {
 std::int64_t __nvqpp_cudaq_state_numberOfQubits(state *obj) {
   return obj->get_num_qubits();
 }
+
+state *__nvqpp_cudaq_state_createFromData_fp64(void *data, std::size_t size) {
+  auto d = reinterpret_cast<std::complex<double> *>(data);
+  auto current = std::vector<std::complex<float>>(d, d + size);
+
+  auto *simulator = cudaq::get_simulator();
+  if (simulator->isSinglePrecision()) {
```

Comment:
@1tnguyen @bmhowe23 Here is where we allow to create a new remote sim state from data that could be coming from a simulation state from the host machine and have different precision.